### ISSUE TITLE
Should `watchPromise` support a remote watcher?

### ISSUE THREAD

**Issue by gibson042** [Agoric/Agoric#7162]
## Describe the bug
`watchPromise(p, remoteWatcher)` seems not to work because https://github.com/Agoric/agoric-sdk/blob/dd7ef095023875d2dfb0191b40b0d8d33c44bb4c/packages/swingset-liveslots/src/watchedPromises.js#L183-L184 rejects watcher vrefs like "o-100" as non-virtual and non-durable.

## Expected behavior
Unclear, but the current error message and behavior disagree with each other.

**Comment by warner** [comment]
I think it'd be pretty easy to make it accomodate remote watchers: change it to something like:
```js
const p = parseVatSlot(watcherVref);
assert(p.type === 'object' && (p.virtual || p.durable || !p.allocatedByVat), '..');
```

That said, it would be expanding the uses (and also expanding our responsibilities in the future) to allow remote watchers. It's not something vats ought to be doing: if vatA is registering a watcher, it's because some object in vatA cares about the promise, and if some object in vatB was interested, vatB can register its own watcher (on a promise which is remote, from its point of view).

So the main motivation for allowing this would be for enabling a simpler unit test of the "watch local promise" feature, whose implementation has changed as part of #6694 / #7130 . Since that PR has landed, the real question is whether we're satisfied with the tests in that PR, or if we think we need better ones (which would be enabled by allowing remote watchers).

**Comment by warner** [comment]
@gibson042 said we're pretty satisified with the tests that landed in #7130 , so I think we don't need this, and by not adding it, we're not adding to our future support obligations. Closing as WONTFIX.
